cloudstack/loadbalancer.go

```diff
@@ -1002,9 +1028,106 @@ func (lb *loadBalancer) createLoadBalancerRule(lbRuleName string, service *v1.Se
 		},
 	}
 
+	err = lb.updateLoadBalancerPool(lbRule, service)
+	if err != nil {
+		return nil, fmt.Errorf("updateLoadBalacerPool error: %v", err)
+	}
+
 	return lbRule, nil
 }
 
+func (lb *loadBalancer) updateLoadBalancerPool(lbRule *loadBalancerRule, service *v1.Service) error {
+
+	client, err := lb.getClient()
+	if err != nil {
+		return err
+	}
+
+	listGloboNetworkPoolsParams := &cloudstack.CustomServiceParams{}
+	listGloboNetworkPoolsResponse := globoNetworkPools{}
+
+	_, lbCustomHealthCheckVal := getLabelOrAnnotation(service.ObjectMeta, lbCustomHealthCheck)
+	if !lbCustomHealthCheckVal {
+		return nil
+	}
+
+	listGloboNetworkPoolsParams.SetParam("lbruleid", lbRule.Id)
+	err = client.Custom.CustomRequest("listGloboNetworkPools", listGloboNetworkPoolsParams, &listGloboNetworkPoolsResponse)
+
+	if err != nil {
+		return fmt.Errorf("error list load balancer pools for %v: %v", lbRule.Name, err)
+	}
+
+	if listGloboNetworkPoolsResponse.Count < 1 {
+		return nil
+	}
+
+	updateGloboNetworkPoolsParams := &cloudstack.CustomServiceParams{}
+
+	r := UpdateGloboNetworkPoolResponse{}
+	for idx := range service.Spec.Ports {
+		pool, err := generateGloboNetworkPool(idx, service, listGloboNetworkPoolsResponse.GloboNetworkPools)
+		if pool == (globoNetworkPool{}) {
```

Comment:
I think this would be clearer returning a pointer of `globoNetworkPool` instead, so you can easily make this comparison just firing the `pool == nil`. In addition, we always (except for specific ones) should checking the error before.

Reply:
You're right. I've misplaced both if clauses. Also, changed function return to a pointer. Thanks